utils/logalloc.cc

```diff
@@ -2259,7 +2259,36 @@ allocating_section::guard::~guard() {
     shard_segment_pool.set_emergency_reserve_max(_prev);
 }
 
+void allocating_section::maybe_decay_reserve() {
+    // The decay rate is inversely proportional to the reserve
+    // (every (s_segments_per_decay/_lsa_reserve) allocations).
+    //
```

Comment:
Approach LGTM

Reply:
Thanks. I'll try to add a unit test for the next iteration.